uniffi_bindgen/src/bindings/python/templates/TopLevelFunctionTemplate.py

```diff
@@ -1,20 +1,39 @@
 {%- if func.is_async() %}
 
-def {{ func.name()|fn_name }}({%- call py::arg_list_decl(func) -%}):
+{%- match func.return_type() -%}
+{%- when Some with (return_type) %}
+
+async def {{ func.name()|fn_name }}({%- call py::arg_list_decl(func) -%}) -> "{{ return_type|type_name }}":
```

Comment:
It looks to me like you've moved the check of `return_type` outside of the function just so this first line can be different, with the cost being more duplicated code between each branch - I'm not sure that's the correct trade-off - I think an extra match block just for this line might be better?

The Python docs also imply that the "returns None" case can also be annotated with `-> None` whereas this just declines to annotate that case.

Reply:
> just so this first line can be different

I brought it outside for the first line of the function as well as the `# lift function` cases inside of the function body, essentially reducing duplication of the `{%- match func.return_type() -%}` and cases lines, but increasing duplication of other lines. So I could change it to two match statements (one outside for the first line of the function, one inside for the lift function), or keep them combined at the top. I'm good with it either way, just wanted to confirm with you first which you'd prefer.

> "returns None" case can also be annotated with `-> None`

I think this is a good idea, which I'd extend to the non-async case as well. I was concerned at first that we'd be making explicit something that was implicit in Rust, but I think the Rust compiler requires a return type in a strict enough way that it's not a problem. Thanks for the suggestion!

Reply:
yeah - while this is very subjective, I think "number of duplicated lines" is a reasonable yardstick and my impression is duplicating the control-flow but not the actual template would be smaller than the opposite.